### PR TITLE
Functional PT Savestates and additional debug

### PR DESCRIPTION
**Pull request by GradualSyrup**
Speeds up PT save state loads to be as fast as other fighters, removes their effects, and allows for the saving of the current Pokemon loaded in the state. Still very subtle issues like PT becoming locked out of running on certain save state loads until a later state load, but nothing glaring or that impacts gameplay.

### CODE REVIEW

File: src/training/character_specific/ptrainer.rs [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+    ptrainer_module_accessor: *mut app::BattleObjectModuleAccessor,
+) -> *mut app::BattleObjectModuleAccessor {
+    let pokemon_object_id = LinkModule::get_node_object_id(
+        ptrainer_module_accessor,
+        *WEAPON_PTRAINER_PTRAINER_LINK_NO_POKEMON,
+    );
+    &mut *app::sv_battle_object::module_accessor(pokemon_object_id as u32)
+}
+
+pub unsafe fn check_effect_pokemon_state(
```
Comment (jugeeya) [inline src/training/character_specific/ptrainer.rs:91]: nit: This isn't really a `check`. Usually if something can return something to potentially mutate a hook, we call it `handle`, and also it's about the size. So maybe `handle_pokemon_effect_size`

File: src/training/character_specific/ptrainer.rs [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+    if (is_ptrainer_switch_hash && save_states::is_loading())
+        || Hash40::new("sys_flying_plate") == hash
+    {
+        // Making the size 0 prevents these effects from being displayed. Fixes Pokemon Trainer Angel Platform Effect.
+        return 0.0;
+    }
+    size
+}
+
+pub unsafe fn sound_effect_pokemon_state(hash: Hash40) -> Hash40 {
```
Comment (jugeeya) [inline src/training/character_specific/ptrainer.rs:127]: Similarly here. `handle_pokemon_sound_effect`?

File: src/training/character_specific/ptrainer.rs [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+        return Hash40::new("se_silent");
+    }
+    hash
+}
+
+// Choose which pokemon to switch to!
+static POKEMON_DECIDE_OFFSET: usize = 0x34cdc64;
+
+#[skyline::hook(offset = POKEMON_DECIDE_OFFSET, inline)]
+unsafe fn pokemon_decide_handle(ctx: &mut InlineCtx) {
```
Comment (jugeeya) [inline src/training/character_specific/ptrainer.rs:144]: `handle_pokemon_decide`

File: src/training/character_specific/ptrainer.rs (hunk 1 above)
Comment (GradualSyrup) [reply src/training/character_specific/ptrainer.rs:91]: I prefer `handle_pokemon_effect`, but agree with the sentiment here. It's not really about resizing the effect, and more that that is the easiest way to "remove" it.

**Pull request merged**